### DISCUSSION TITLE
LAB 04.02 task 1. Building datasets

### DISCUSSION THREAD

**Discussion by EstebanMa12** [rramosp/ai4eng.v1#172] [04 - GESTIÓN DE DATOS]
A la hora de extraer los datos con la fecha asignada me aparece un error, creo que hay un error en la base de datos, puesto a que revisando, el dataset no pasa de 2016-01, con el ID asignado 
![image](https://user-images.githubusercontent.com/64288956/180321903-4cf0a8d4-c5c1-486f-a5f3-42926bee7b50.png)
Cuando intento hacer el filtrado me aparece un error, adjunto imagen
![image](https://user-images.githubusercontent.com/64288956/180322035-9c5df8d1-4c9b-4492-9a12-e8b0529d0422.png)

Espero atento a su respuesta
@rramosp @Felipe-RA @raulramosudea

**Comment by Felipe-RA** [comment]
Saludos EstebaMa

No es un problema del dataset. El problema ocurre en la forma que aplicas la función `df.loc[]` para filtrar las fechas, pues utilizas el timestamp como si fuese un label (de ahí que surja el error `KeyError`, pues no existe un label 2016-02-10 para el indice del dataframe)

Recuerda que `timestamp` es una columna, no un indice.

También, ten en cuenta que `df.loc[]` recibe también un array de booleanos, por lo que puedes incluir una **condición** para filtrar tus datos según la fecha que necesitas. Te sugiero ese acercamiento.

Echale un vistazo a la documentación de la función: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html

**Comment by EstebanMa12** [comment]
Dale, muchas gracias

El vie., 22 de julio de 2022 11:25 a. m., Felipe-RA <
***@***.***> escribió:

> Saludos EstebaMa
>
> No es un problema del dataset. El problema ocurre en la forma que aplicas
> la función df.loc[] para filtrar las fechas, pues utilizas el timestamp
> como si fuese un label (de ahí que surja el error KeyError, pues no
> existe un label 2016-02-10 para el indice del dataframe)
>
> Recuerda que timestamp es una columna, no un indice.
>
> También, ten en cuenta que df.loc[] recibe también un array de booleanos,
> por lo que puedes incluir una *condición* para filtrar tus datos según la
> fecha que necesitas. Te sugiero ese acercamiento.
>
> Echale un vistazo a la documentación de la función:
> https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/172#discussioncomment-3207671>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APKPRPAUB22KGYKHUOIAXDDVVLDP5ANCNFSM54JEFNUQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
"La información aquí contenida es para uso exclusivo de la persona o 
entidad de destino. Está estrictamente prohibida su utilización, copia, 
descarga, distribución, modificación y/o reproducción total o parcial, sin 
el permiso expreso de Universidad de Antioquia, pues su contenido puede ser 
de carácter confidencial y/o contener material privilegiado. Si usted 
recibió esta información por error, por favor contacte en forma inmediata a 
quien la envió y borre este material de su computador. Universidad de 
Antioquia no es responsable por la información contenida en esta 
comunicación, el directo responsable es quien la firma o el autor de la 
misma."

-- 
UdeA
